### ISSUE TITLE
🌳 Task: Decision Tree Training & Evaluation

### ISSUE THREAD

**Issue by Minh08012005** [Minh08012005/Heart-Disease-Prediction-ML-Group-6#6]
**📅 Deadline:** 4/5/2026
**👥 Assignees:** @Minh08012005 , @vantuann205 , @DHieu2105 , @punnohoang 
**🔗 Chi tiết:** Xem file https://github.com/Minh08012005/Heart-Disease-Prediction-ML-Group-6/blob/main/task_V2.md trong repo

---

### 🎯 Mục tiêu

Mỗi người tự tạo 1 notebook riêng để train, tuning, và đánh giá Decision Tree, sau đó so sánh với sklearn.

---

### 📝 Công việc cụ thể

| # | Công việc | Mô tả |
|---|-----------|-------|
| 1 | **Feature Selection** | Dùng correlation để chọn features quan trọng (bỏ features nhiễu) |
| 2 | **Train cơ bản** | Train Custom DT với tham số mặc định (max_depth=5) |
| 3 | **Đánh giá** | Tính Accuracy, Precision, Recall, F1-Score, Confusion Matrix |
| 4 | **Hyperparameter Tuning** | Thử các bộ (max_depth, min_samples) khác nhau để tìm best |
| 5 | **So sánh sklearn** | So sánh Custom DT vs sklearn DecisionTreeClassifier cùng tham số |
| 6 | **Kết luận** | Nhận xét kết quả, đề xuất cải thiện |

---

### 👥 Phân công nhánh

| Thành viên | Nhánh Git | Notebook |
|------------|-----------|----------|
| @Minh | `minh/dt-training` | `notebooks/04_DT_Minh.ipynb` |
| @Hiếu | `hieu/dt-training` | `notebooks/04_DT_Hieu.ipynb` |
| @Phong | `phong/dt-training` | `notebooks/04_DT_Phong.ipynb` |
| @Tuân | `tuan/dt-training` | `notebooks/04_DT_Tuan.ipynb` |

---

### 📋 Quy trình Git

```bash
git checkout main && git pull origin main
git checkout -b ten-ban/dt-training
# Tạo notebook theo hướng dẫn trong task_V2.md
git add notebooks/04_DT_TenBan.ipynb
git commit -m "Hoàn thành DT training"
git push origin ten-ban/dt-training
# Tạo Pull Request vào main, gán @Minh review
```

---

### ✅ Deliverables

- [ ] Notebook `.ipynb` hoàn chỉnh (11 cells)
- [ ] Best accuracy sau tuning
- [ ] So sánh với sklearn
- [ ] Pull Request vào `main`

---
